src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,26 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types';
+import { getUser } from '../../api';
+import { TodoContext } from '../../TodoContext';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const { selectedTodo, setSelectedTodo } = useContext(TodoContext);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId)
+        .then(setUser);
+    }
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
```

Comment:
It's not a good idea. Add state for `isLoading`, the same as you did it in `App` component.
What if you will get an empty response for `user`? You will see loader all the time